File: coder_byte.js
// Have the function SimpleAdding(num) add up all the numbers from 1 to num. For the test cases, the parameter num will be any number from 1 to 1000. 

function SimpleAdding(num) {
  var total = num;
  var count = 1;
  
  for(var i = 0; i < num; ++i){ 
    total = total + num-count;
    count = count + 1;
  }
  return total;     
}

// Have the function AlphabetSoup(str) take the str string parameter being passed and return the string with the letters in alphabetical order (ie. hello becomes ehllo). Assume numbers and punctuation symbols will not be included in the string. 

function AlphabetSoup(str) { 

  newArray = str.split("");
  newArray.sort();
  newStr = newArray.join("");
  return newStr;       
}

// Have the function VowelCount(str) take the str string parameter being passed and return the number of vowels the string contains (ie. "All cows eat grass" would return 5). Do not count y as a vowel for this challenge. 

function VowelCount(str) {
  var vowelArray = ["a","e","i","o","u"];  
  var letterArray = str.split(""); 
  var vowelCount = 0;
  
  for(var i = 0; i < letterArray.length; ++i){ 
    if(vowelArray.indexOf(letterArray[i]) === -1){
      vowelCount = vowelCount;
    } else {
      vowelCount = vowelCount + 1;
    }
  }
  return vowelCount;
}

File: coder_byte.rb
# Have the function FirstReverse(str) take the str parameter being passed and return the string in reversed order.

def FirstReverse(str)
  str.reverse!
  return str         
end


# Have the function FirstFactorial(num) take the num parameter being passed and return the factorial of it (ie. if num = 4, return (4 * 3 * 2 * 1)). For the test cases, the range will be between 1 and 18. 
def FirstFactorial(num)
  (1..num).reduce(:*)
end


# Have the function Palindrome(str) take the str parameter being passed and return the string true if the parameter is a palindrome, (the string is the same forward as it is backward) otherwise return the string false. For example: "racecar" is also "racecar" backwards. Punctuation and numbers will not be part of the string. 

def Palindrome(str)
  no_spaces = str.gsub(" ","")
  if no_spaces == no_spaces.reverse
    return true
  else
    return false
  end      
end

# Have the function LetterCapitalize(str) take the str parameter being passed and capitalize the first letter of each word. Words will be separated by only one space. 

def LetterCapitalize(str)
  word_array = str.split(" ")
  upper = word_array.map do |word|
    word.capitalize
  end
  new_str = upper.join(" ")
  return new_str     
end

# Have the function CheckNums(num1,num2) take both parameters being passed and return the string true if num2 is greater than num1, otherwise return the string false. If the parameter values are equal to each other then return the string -1. 

def CheckNums(num1,num2)
  if num1 > num2
    return false
  elsif num1 < num2
    return true
  elsif num1 == num2
    return "-1"
    end
end
   



